### ISSUE TITLE
Datetime field shows 'current date & time' in Index view, even though Database value is NULL (if using a default() value)

### ISSUE THREAD

**Issue by YasamDevelopment** [laravel/nova-issues#3366]
- Laravel Version: 7.30.4
- Nova Version: 3.24.0
- PHP Version: 7.4.16
- Database Driver & Version: mysql 8
- Operating System and Version: Ubuntu 20.04 (LTS) x64
- Browser type and version: Chrome 89.0.4389.128 & Safari 14.0

### Description:
- If you have a Datetime value in your DB that is NULL, it shows up as something like: '28.04.2021 12:19:51' in the "Index" view of the resource (if and only if you use the `->default()` on the DateTime field (see below)).
- If you click to "View" that resource, that datetime shows correctly as a dash (-).
- If you click "Edit" for that resource, that datetime correctly is non-existent.
- The problem is only seen when displaying the datetime of the resource on the "Index" view.

#### That's the bug, but in case you're wondering how I even see that bug since I'm using a `->default()`, on that DateTime field (and you might think the DB would always contain a non-null value for that datetime):
- I have a "published" field (boolean) & "published_at" field (datetime).
  - In Nova, the 'published_at' is a standard DateTime field, with a `->default(now())`. 
- When Creating a new Resource, in the case that the user, does NOT check published, (and leaves the date as the default date), then I (in code), save the 'published_at' datetime as NULL to the DB.
- In this way, that DateTime gets stored as NULL in the DB, and from there you can easily see this bug by viewing that resource on Nova's "Index" view for that resource.

### Detailed steps to reproduce the issue on a fresh Nova installation:
- Create a new Resource, and give it a Datetime field. Something like this:
```
DateTime::make('Published At')
                ->format('DD.MM.YYYY HH:mm:ss')
                ->pickerDisplayFormat('d.m.Y H:i:S')
                ->default(now()),
```
- (make sure to include the `->default(now())` part
- Create a new Resource (give it any date)
- After creation, go to DB & make the date NULL  *See my explanation above for valid use case of saving a NULL date
- View the resource in Nova's Index view (you will see the current datetime written - even though the DB value is NULL)
- (If you view the same resource in Nova's Detail or Edit view, you will see that it displays correctly.)

**Comment by crynobone** [comment]
Use the following to limit just for creating context:

```php
DateTime::make('Verified At', 'email_verified_at')
    ->default(function ($request) {
        if ($request->isCreateOrAttachRequest()) {
            return now();
        }
    }),
```

**Comment by crynobone** [comment]
This has been fixed with [v3.25.0 release](https://nova.laravel.com/releases/3.25.0).
